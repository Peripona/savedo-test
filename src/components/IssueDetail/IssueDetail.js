import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './IssueDetail.module.css';
import { connect } from 'react-redux';

class IssueDetail extends Component {
  render() {
    const { prop = 'Hello' } = this.props;
    return (
      <div className={styles.IssueDetail}>
        <span>{prop}</span>
        <h1>{this.props.match.params.issueId}</h1>
        <h2>{this.props.selected.number}</h2>
      </div>
    );
  }
}

IssueDetail.propTypes = {
  prop: PropTypes.string.isRequired,
};

const mapsStateToProps = state => {
  const { selected } = state.issues;
  return {
    selected,
  };
};

const IssueDetailConnected = connect(mapsStateToProps)(IssueDetail);

export { IssueDetail, IssueDetailConnected };
