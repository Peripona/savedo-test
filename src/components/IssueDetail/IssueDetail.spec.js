import React from 'react';
import { shallow } from 'enzyme';
import { IssueDetail } from './IssueDetail';

describe('<IssueDetail />', () => {
  const build = passedProps => {
    const defaultProps = {
      match: {
        params: {
          issueId: 123,
        },
      },
      selected: 123,
    };
    return shallow(<IssueDetail {...defaultProps} {...passedProps} />);
  };

  it('should render', () => {
    const component = build({ prop: 'Testing' });
    expect(component).toMatchSnapshot();
  });
});
