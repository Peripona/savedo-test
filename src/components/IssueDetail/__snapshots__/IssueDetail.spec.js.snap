// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueDetail /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IssueDetail
    match={
      Object {
        "params": Object {
          "issueId": 123,
        },
      }
    }
    prop="Testing"
    selected={123}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Testing
        </span>,
        <h1>
          123
        </h1>,
        <h2 />,
      ],
      "className": "IssueDetail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Testing",
        },
        "ref": null,
        "rendered": "Testing",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 123,
        },
        "ref": null,
        "rendered": 123,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Testing
          </span>,
          <h1>
            123
          </h1>,
          <h2 />,
        ],
        "className": "IssueDetail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Testing",
          },
          "ref": null,
          "rendered": "Testing",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 123,
          },
          "ref": null,
          "rendered": 123,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
